code/modules/reagents/machinery/reagentgrinder.dm

```diff
@@ -0,0 +1,632 @@
+				else
+					if (O.reagents != null && O.reagents.has_reagent(NUTRIMENT))
+						beaker.reagents.add_reagent(r_id, min(round(O.reagents.get_reagent_amount(NUTRIMENT)*abs(amount)), space))
+						O.reagents.remove_reagent(NUTRIMENT, min(O.reagents.get_reagent_amount(NUTRIMENT), space))
+
+			else
+				O.reagents.trans_id_to(beaker, r_id, min(amount, space))
+
+			if (beaker.reagents.total_volume >= beaker.reagents.maximum_volume)
+				break
+
+		if(O.reagents.reagent_list.len == 0)
+			remove_object(O)
+
+	//Sheets
+	for (var/obj/item/stack/sheet/O in holdingitems)
+		var/allowed = get_allowed_by_id(O)
+		if (beaker.reagents.total_volume >= beaker.reagents.maximum_volume)
+			break
+		for(var/i = 1; i <= round(O.amount, 1); i++)
+			for (var/r_id in allowed)
+				var/space = beaker.reagents.maximum_volume - beaker.reagents.total_volume
+				var/amount = allowed[r_id]
+				beaker.reagents.add_reagent(r_id,min(amount, space))
+				if (space < amount)
+					break
+			if (i == round(O.amount, 1))
+				remove_object(O)
+				break
+
+	//xenoarch
+	for(var/obj/item/weapon/rocksliver/O in holdingitems)
+		if (beaker.reagents.total_volume >= beaker.reagents.maximum_volume)
+			break
+		var/allowed = get_allowed_by_id(O)
+		for (var/r_id in allowed)
+			var/space = beaker.reagents.maximum_volume - beaker.reagents.total_volume
+			var/amount = allowed[r_id]
+			beaker.reagents.add_reagent(r_id,min(amount, space), O.geological_data)
+
+			if (beaker.reagents.total_volume >= beaker.reagents.maximum_volume)
+				break
+		remove_object(O)
+
+	//Everything else - Transfers reagents from it into beaker
+	for (var/obj/item/weapon/reagent_containers/O in holdingitems)
+		if (beaker.reagents.total_volume >= beaker.reagents.maximum_volume)
+			break
+		var/amount = O.reagents.total_volume
+		O.reagents.trans_to(beaker, amount)
+		if(!O.reagents.total_volume)
+			remove_object(O)
+
+	//All other generics
+	for (var/obj/item/O in holdingitems)
+		if (beaker.reagents.total_volume >= beaker.reagents.maximum_volume)
+			break
+		var/allowed = get_allowed_by_id(O)
+		for (var/r_id in allowed)
+			var/space = beaker.reagents.maximum_volume - beaker.reagents.total_volume
+			var/amount = allowed[r_id]
+			if (amount == 0)
+				if (O.reagents != null && O.reagents.has_reagent(r_id))
+					beaker.reagents.add_reagent(r_id,min(O.reagents.get_reagent_amount(r_id), space))
+			else
+				beaker.reagents.add_reagent(r_id,min(amount, space))
+
+			if (beaker.reagents.total_volume >= beaker.reagents.maximum_volume)
+				break
+		remove_object(O)
+
+/obj/item/weapon/reagent_containers/glass/mortar
+	name = "mortar"
+	desc = "This is a reinforced bowl, used for crushing reagents. Ooga booga Rockstop."
+	icon = 'icons/obj/food.dmi'
+	icon_state = "mortar"
+	flags = FPRINT  | OPENCONTAINER
+	volume = 50
+	amount_per_transfer_from_this = 5
+	//We want the all-in-one grinder audience
+
+	var/list/blend_items = list (
+		/obj/item/stack/sheet/metal           = list(IRON,20),
```

Comment:
No it wouldn't
I don't want to scroll through over 2000 changed lines to review one (1) refactored machine